src/frontend/lib/utils.ts

```diff
@@ -50,30 +46,40 @@ export type LocationStatus = 'searching' | 'improving' | 'good' | 'error';
 //   return isNaN(major) ? 0 : major;
 // }
 
-// export function getLocationStatus({
-//   position,
-//   provider,
-//   permission,
-//   error,
-// }: Type): LocationStatus {
-//   const precision = position && position.coords.accuracy;
-//   const gpsUnavailable = provider && !provider.gpsAvailable;
-//   const locationServicesDisabled =
-//     provider && !provider.locationServicesEnabled;
-//   const noPermission = permission && permission !== "granted";
-//   const positionStale =
-//     position && Date.now() - position.timestamp > STALE_TIMEOUT;
-//   if (error || gpsUnavailable || locationServicesDisabled || noPermission)
-//     return "error";
-//   else if (positionStale) return "searching";
-//   else if (
-//     typeof precision === "number" &&
-//     Math.round(precision) <= GOOD_PRECISION
-//   )
-//     return "good";
-//   else if (typeof precision === "number") return "improving";
-//   return "searching";
-// }
+// If the current position on the app state is more than 60 seconds old then we
+// consider it stale and show that the GPS is searching for a new position
+const STALE_TIMEOUT = 60 * 1000; // 60 seconds
+// // If the precision is less than 10 meters then we consider this to be a "good
+// // position" and we change the UI accordingly
+const GOOD_PRECISION = 10; // 10 meters
+
+export type LocationStatus = 'searching' | 'improving' | 'good' | 'error';
+
+export function getLocationStatus({
+  location,
+  providerStatus,
+}: {
+  location?: LocationObject;
+  providerStatus?: LocationProviderStatus;
+}): LocationStatus {
+  const gpsNotAvailable = !providerStatus?.gpsAvailable;
+  const locationServicesDisabled = !providerStatus?.locationServicesEnabled;
```

Comment:
nit: would call `gpsAvailable` and `locationServicesEnabled` to better match location services API.

Reply:
addressed via https://github.com/digidem/CoMapeo-mobile/pull/212/commits/12a910a06877512719c44ecc0251d9ab40d38f4f